keylime/src/crypto.rs

```diff
@@ -607,6 +644,18 @@ fn pkey_pub_from_priv(
             .map_err(CryptoError::RSAFromComponents)?;
             PKey::from_rsa(rsa).map_err(CryptoError::PKeyFromRSAError)
         }
+        Id::EC => {
+            let ec_key = privkey
+                .ec_key()
+                .map_err(CryptoError::ECGetPrivateKeyError)?;
+
+            let pubkey =
+                EcKey::from_public_key(ec_key.group(), ec_key.public_key())
+                    .map_err(CryptoError::ECKeyFromPublicPointError)?;
+
+            Ok(PKey::from_ec_key(pubkey)
```

Comment:
I guess you don't need to wrap this with `Ok(...?)`.

Reply:
Thanks!